lib/nats.js

```diff
@@ -797,6 +797,31 @@ Client.prototype.processInbound = function() {
 	if (client.checkTLSMismatch() === true) {
 	  return;
 	}
+
+  // Always try to read the connect_urls from info
+  if(client.info.connect_urls && client.info.connect_urls.length > 0) {
+    // don't add duplicates
+    var known = [];
+    client.servers.forEach(function(server) {
+      known.push(server.url.href);
+    });
+    // add new ones
+    var toAdd = [];
+    client.info.connect_urls.forEach(function(server) {
+      var u = 'nats://' + server;
+      if(known.indexOf(u) === -1) {
+        toAdd.push(new Server(url.parse(u)));
+      }
+    });
+
+    if(toAdd.length > 0) {
+      shuffle(toAdd);
```

Comment:
If I am not mistaken, this client has a `noRandomize` option. If so, you should shuffle the incoming array only if `noRandomize === false`.